### PR TITLE
feat: add arm64 binary support

### PR DESCRIPTION
**Pull request by jeluard**
**Motivation**

Add `arm64` binary support. Those binaries are generated and published during a release publication.
Tested on `linux arm64`, works fine. A test binary can be found [here](https://github.com/ChainSafe/lodestar/actions/runs/8847869356/artifacts/1450708201)

### CODE REVIEW

File: .github/workflows/binaries.yml [hunk 1]
```diff
@@ -19,19 +19,29 @@ jobs:
...
+          - os: lodestar-arm64-runner
+            platform: linux
+            arch: arm64
     runs-on: ${{matrix.os}}
     steps:
       - uses: actions/checkout@v4
+      - name: Install yarn
+        if: matrix.arch == 'arm64'
+        run: |-
+          curl -fsSL --create-dirs -o $HOME/bin/yarn \
```
Comment (nflaig) [inline .github/workflows/binaries.yml:35]: why is this required for arm64?

**Review by nflaig — COMMENTED** [review]
Did you test the binary on one of our arm64 servers?

File: .github/workflows/binaries.yml (hunk 1 above)
Comment (jeluard) [reply .github/workflows/binaries.yml:35]: That's a limitation of the current `arm64` runner image

File: .github/workflows/binaries.yml [hunk 2]
```diff
@@ -19,19 +19,31 @@ jobs:
...
+            platform: linux
+            arch: amd64
+          - os: lodestar-arm64-runner
+            platform: linux
+            arch: arm64
     runs-on: ${{matrix.os}}
     steps:
       - uses: actions/checkout@v4
+      # Install yarn for arm64 as currently it's not part of the provided image
+      # See https://github.com/github-early-access/arm-runners-beta/issues/5
```
Comment (nflaig) [inline .github/workflows/binaries.yml]: I don't have access to this link, just getting a 404

Comment (jeluard) [reply .github/workflows/binaries.yml]: Right, you have to be part of the private beta. Keeping it here for future reference.
Once it moves out of beta I would expect `yarn` to be installed by default.

**Review by nflaig — APPROVED** [review]
LGTM

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.19.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.19.0) :tada:
